dbcon/execplan/treenode.h

```diff
@@ -974,8 +989,15 @@ inline long double TreeNode::getLongDoubleVal()
         case CalpontSystemCatalog::DECIMAL:
         case CalpontSystemCatalog::UDECIMAL:
         {
-            // this may not be accurate. if this is problematic, change to pre-calculated power array.
-            return (long double)(fResult.decimalVal.value / pow((long double)10, fResult.decimalVal.scale));
+            if (fResultType.colWidth == datatypes::MAXDECIMALWIDTH)
+            {
+                return static_cast<long double>(fResult.decimalVal);
+            }
+            else
+            {
+                // this may not be accurate. if this is problematic, change to pre-calculated power array.
+                return (long double)(fResult.decimalVal.value / pow((long double)10, fResult.decimalVal.scale));
```

Comment:
I would like to make this a method of the Decimal class but I think you would like to change it globally in a separate patch.

Reply:
I agree, this change should be outside the scope of this patch.